ui_src/src/components/createStationForm/index.js

```diff
@@ -62,70 +65,74 @@ const storageOptions = [
     }
 ];
 
-const CreateStationForm = ({ createStationFormRef }) => {
-    const [getStartedState, getStartedDispatch] = useContext(GetStartedStoreContext);
+const CreateStationForm = ({ createStationFormRef, getStartedStateRef, finishUpdate, updateFormState, getStarted, setLoading }) => {
+    const history = useHistory();
     const [creationForm] = Form.useForm();
     const [allowEdit, setAllowEdit] = useState(true);
+    const [actualPods, setActualPods] = useState(null);
+    const [retentionType, setRetentionType] = useState(retanionOptions[0].value);
+    const [storageType, setStorageType] = useState(storageOptions[0].value);
 
     useEffect(() => {
-        checkState();
-        createStationFormRef.current = onFinish;
+        getOverviewData();
+        if (getStarted && getStartedStateRef?.completedSteps > 0) setAllowEdit(false);
+        if (getStarted && getStartedStateRef?.formFieldsCreateStation?.retention_type) setRetentionType(getStartedStateRef.formFieldsCreateStation.retention_type);
     }, []);
 
-    const checkState = () => {
-        if (getStartedState?.formFieldsCreateStation?.name) {
-            setAllowEdit(false);
-            getStartedDispatch({ type: 'SET_NEXT_DISABLE', payload: false });
-        } else {
-            setAllowEdit(true);
+    useEffect(() => {
+        createStationFormRef.current = onFinish;
+    }, [createStationFormRef, getStartedStateRef]);
```

Comment:
Why do you need these arguments and not always when component DidMount?

Reply:
You right. Fixed it.